test_install.py fails when testing a tagged version

test_install.py includes a test that assumes that the default version of desimodel data is trunk.  However, PR #179 updated that to install the matching tagged version by default so that the user doesn't have to manually specify that tag as well.  The result is that test pass on non-tagged versions of desimodel (which will install trunk by default), but it fails when testing a tag, e.g.

```
git checkout 0.19.3
pytest py/desimodel/test/test_install.py
...
E                   AssertionError: expected call not found.
E                   Expected: Popen(['svn', 'export', 'https://desi.lbl.gov/svn/code/desimodel/trunk/data'], stderr=-1, stdout=-1)
E                   Actual: Popen(['svn', 'export', 'https://desi.lbl.gov/svn/code/desimodel/tags/0.19.3/data'], stdout=-1, stderr=-1)
E                   
E                   pytest introspection follows:
E                   
E                   Args:
E                   assert (['svn', 'exp....19.3/data'],) == (['svn', 'exp...trunk/data'],)
E                     
E                     At index 0 diff: ['svn', 'export', 'https://desi.lbl.gov/svn/code/desimodel/tags/0.19.3/data'] != ['svn', 'export', 'https://desi.lbl.gov/svn/code/desimodel/trunk/data']
E                     Use -v to get more diff

py/desimodel/test/test_install.py:137: AssertionError
```

The `test_svn_export` test in the same file already tests the tags vs. trunk behavior for various settings of `desimodel.__version__`, but we need to update the `desimodel.test.test_install.test_install` function to not trip when testing tagged versions of desimodel.

--- a/py/desimodel/test/test_install.py
+++ b/py/desimodel/test/test_install.py
@@ -7,7 +7,7 @@ import shutil
 from tempfile import mkdtemp
 from subprocess import CalledProcessError
 import unittest
-from unittest.mock import patch, MagicMock, call
+from unittest.mock import patch, call
 from ..install import default_install_dir, assert_svn_exists, svn_export, install, add_files_to_record
 import desimodel
 
@@ -154,14 +154,36 @@ class TestInstall(unittest.TestCase):
                 pass
             with patch('desimodel.install.assert_svn_exists'):
                 with patch('os.chdir'):
-                    with patch('desimodel.install.Popen') as Popen:
-                        proc = Popen.return_value
-                        proc.communicate.return_value = ('Mock stdout', 'Mock stderr')
-                        proc.returncode = 1
-                        with self.assertRaises(RuntimeError) as e:
-                            install(desimodel='/opt/desimodel')
-                        self.assertEqual(str(e.exception), "Mock stderr")
-                    Popen.assert_called_once_with(['svn', 'export', 'https://desi.lbl.gov/svn/code/desimodel/trunk/data'], stderr=-1, stdout=-1)
+                    #
+                    # Test with version explicitly set.
+                    #
+                    for v in ('0.19.3', '0.20.1.dev1234', 'branches/test-0.19', 'trunk'):
+                        svn_command = svn_export(v)
+                        with patch('desimodel.install.Popen') as Popen:
+                            proc = Popen.return_value
+                            proc.communicate.return_value = ('Mock stdout', 'Mock stderr')
+                            proc.returncode = 1
+                            with self.assertRaises(RuntimeError) as e:
+                                install(desimodel='/opt/desimodel', version=v)
+                            self.assertEqual(str(e.exception), "Mock stderr")
+                        Popen.assert_called_once_with(svn_command, stderr=-1, stdout=-1)
+                    #
+                    # Test with version obtained from desimodel.__version__.
+                    #
+                    for v in ('0.19.3', '0.20.1.dev1234'):
+                        desimodel.__version__ = v
+                        svn_command = svn_export()
+                        with patch('desimodel.install.Popen') as Popen:
+                            proc = Popen.return_value
+                            proc.communicate.return_value = ('Mock stdout', 'Mock stderr')
+                            proc.returncode = 1
+                            with self.assertRaises(RuntimeError) as e:
+                                install(desimodel='/opt/desimodel')
+                            self.assertEqual(str(e.exception), "Mock stderr")
+                        Popen.assert_called_once_with(svn_command, stderr=-1, stdout=-1)
+                    #
+                    # Test dry_run mode.
+                    #
                     with patch('builtins.print') as mock_print:
                         with patch('desimodel.install.default_install_dir') as mock_dir:
                             mock_dir.return_value = 'foo'
@@ -170,4 +192,4 @@ class TestInstall(unittest.TestCase):
                             os.makedirs(desimodel_code)
                             added = install(desimodel=desimodel_code, version=version, svn_checkout=False, dry_run=True)
                             mock_print.assert_has_calls([call(f'Installing desimodel data tags/{version} to {desimodel_code}'),
-                                                        call(f'Dry run, would have run "svn export https://desi.lbl.gov/svn/code/desimodel/tags/{version}/data"')])
+                                                         call(f'Dry run, would have run "svn export https://desi.lbl.gov/svn/code/desimodel/tags/{version}/data"')])

